src/module.ts

```diff
@@ -1,6 +1,36 @@
 import { AppPlugin } from '@grafana/data';
+import { DataQuery, DataSourceRef } from '@grafana/schema';
 import { AppPluginSettings } from '@shared/types/AppPluginSettings';
 
 import { App } from './app/App';
 
-export const plugin = new AppPlugin<AppPluginSettings>().setRootPage(App);
+// TODO: replace with type from grafana/runtime
+const QueryToAppPlugin = 'grafana/query/query-to-app-plugin';
+
+// TODO: replace with type from grafana/runtime
+type QueryToAppPluginContext = {
+  datasource: DataSourceRef;
+  query: DataQuery;
+};
+
+export const plugin = new AppPlugin<AppPluginSettings>()
+  .addLink({
+    targets: [QueryToAppPlugin],
+    title: 'Explore with Profiles app',
+    description: 'Link to Explore Profiles app',
+    path: '/a/grafana-pyroscope-app/profiles-explorer',
+    configure(context: QueryToAppPluginContext | undefined) {
+      if (!context || !context.query || !context.datasource) {
+        return undefined;
+      }
+
+      if (context.datasource.type !== 'grafana-pyroscope-datasource') {
+        return undefined;
+      }
+
+      return {
+        path: `/a/grafana-pyroscope-app/profiles-explorer?var-dataSource=${context.datasource.uid}`,
```

Comment:
Should we try to determine the service name and link to the "Profile types" view?

Reply:
Yes! Ultimately we will determine service name, profile and labels in follow-up PRs coming soon after this one. This version is bare bones to get started with all plugins 👍

Reply:
I'll update the PR to include the service name and from/to if it's provided